travertino/src/travertino/properties/shorthand.py

```diff
@@ -50,10 +65,91 @@ def __set__(self, style, value):
                 f"tuple."
             )
 
-    def __delete__(self, style):
-        with style.batch_apply():
-            for name in self.property_names:
-                del style[name]
 
-    def is_set_on(self, style):
-        return any(name in style for name in self.property_names)
+class composite_property(shorthand_property):
+    def __init__(self, optional, required):
+        """Define a property attribute that proxies for an arbitrary set of properties.
+
+        :param optional: The names of aliased properties that are optional in
+           assignment. Order is irrelevant, unless the same value is valid for more than
+           one property; in that case, values are assigned by priority of which one
+           could apply to the fewest properties, and when that ties, they're assigned in
+           order.
+        :param required: Which properties, if any, are required when setting this
+            property. In assignment, these must be specified last and in order.
+        """
+        self.optional = optional
+        self.required = required
+        self.property_names = self.optional + self.required
+        self.min_num = len(self.required)
+        self.max_num = len(self.required) + len(self.optional)
+
+    def __set__(self, style, composite_value):
+        if composite_value is self:  # pragma: no-cover-if-lt-py310
+            # This happens during autogenerated dataclass __init__ when no value is
+            # supplied.
+            return
+
+        if not self.min_num <= len(composite_value) <= self.max_num:
+            raise TypeError(
+                f"Composite property '{self.name}' must be set with at least "
+                f"{self.min_num} and no more than {self.max_num} values."
```

Comment:
As an error, this is 100% correct... but not especially helpful to an end user. It might be more helpful to elaborate the specific attributes that must be provided (.e.g, "Composite property 'font' must provide a value for 'font_size' and 'font_family', and may also provide a value for 'font_style', 'font_variant' and 'font_weight'.)